crates/lair_keystore_api/src/lair_api/hello.rs

```diff
@@ -0,0 +1,87 @@
+use super::*;
+
+/// Initiate communication with the target lair instance.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LairApiReqHello {
+    /// msg id to relate request / response.
+    pub msg_id: Arc<str>,
+
+    /// random data for server identity verification.
+    pub nonce: BinData,
+}
+
+impl LairApiReqHello {
+    /// Make a new server info request
+    pub fn new(nonce: BinData) -> Self {
+        Self {
+            msg_id: new_msg_id(),
+            nonce,
+        }
+    }
+}
+
+impl std::convert::TryFrom<LairApiEnum> for LairApiReqHello {
+    type Error = one_err::OneErr;
+
+    fn try_from(e: LairApiEnum) -> Result<Self, Self::Error> {
+        if let LairApiEnum::ReqHello(s) = e {
+            Ok(s)
+        } else {
+            Err(format!("Invalid response type: {:?}", e).into())
+        }
+    }
+}
+
+impl AsLairCodec for LairApiReqHello {
+    fn into_api_enum(self) -> LairApiEnum {
+        LairApiEnum::ReqHello(self)
+    }
+}
+
+/// The hello response from the target lair instance.
+/// This data allows us to verify we are speaking to our expected target.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LairApiResHello {
+    /// msg id to relate request / response.
+    pub msg_id: Arc<str>,
```

Comment:
Any reason these need to be an Arc?

Reply:
They are cloned / used as hash map key, etc.

Reply:
Seems like these 3 fields together might be under 100 bytes. Wonder if cloning 100 bytes is significantly more overhead than 3 Arcs.

Reply:
arcs only add 2 usizes of overhead, a single clone of an arc almost always validates its usage